src/viewProviders/flinkArtifacts.test.ts

```diff
@@ -83,79 +86,288 @@ describe("FlinkArtifactsViewProvider", () => {
+        const mockError = new ResponseError(new Response("some other 4xx err", { status: 431 }));
+        stubbedLoader.getFlinkArtifacts.rejects(mockError);
 
-    it("should handle non-401 errors without firing auth session invalidated", async () => {
-      const windowWithProgressStub = sandbox
-        .stub(window, "withProgress")
-        .callsFake((_, callback) => {
-          // Call the callback immediately with a resolved promise
-          const mockProgress = {} as Progress<unknown>;
-          const mockToken = {} as CancellationToken;
-          return Promise.resolve(callback(mockProgress, mockToken));
-        });
-
-      const authInvalidatedFireStub = sandbox.stub(ccloudAuthSessionInvalidated, "fire");
-      const resource = TEST_CCLOUD_FLINK_COMPUTE_POOL;
-      viewProvider["resource"] = resource;
+        await assert.rejects(async () => {
+          await viewProvider.refresh();
+        }, mockError);
 
-      const stubbedLoader: sinon.SinonStubbedInstance<CCloudResourceLoader> =
-        getStubbedCCloudResourceLoader(sandbox);
+        sinon.assert.calledOnce(logErrorStub);
+        sinon.assert.calledWith(logErrorStub, mockError, "Failed to load Flink artifacts");
 
-      const serverError = createResponseError(500, "Internal Server Error", "test");
-      stubbedLoader.getFlinkArtifacts.rejects(serverError);
+        sinon.assert.calledOnce(showErrorNotificationStub);
+        sinon.assert.calledWith(
+          showErrorNotificationStub,
+          "Failed to load Flink artifacts due to an unexpected error.",
+        );
+      });
+      it("should handle 4xx HTTP errors with appropriate message", async () => {
+        // Use a 431 status, which is not explicitly handled in the switch statement,
+        // so it should use the default message: "Failed to load Flink artifacts due to an unexpected error."
+        const mockError = new ResponseError(new Response("some other 4xx err", { status: 431 }));
+        stubbedLoader.getFlinkArtifacts.rejects(mockError);
+
+        await assert.rejects(async () => {
+          await viewProvider.refresh();
+        }, mockError);
+
+        sinon.assert.calledOnce(logErrorStub);
+        sinon.assert.calledWith(logErrorStub, mockError, "Failed to load Flink artifacts");
 
-      await assert.rejects(async () => {
-        await viewProvider.refresh();
+        sinon.assert.calledOnce(showErrorNotificationStub);
+        sinon.assert.calledWith(
+          showErrorNotificationStub,
+          "Failed to load Flink artifacts due to an unexpected error.",
+        );
       });
+      it("should handle 4xx HTTP errors with appropriate message", async () => {
+        // Use a 431 status, which is not explicitly handled in the switch statement,
+        // so it should use the default message: "Failed to load Flink artifacts due to an unexpected error."
+        const mockError = new ResponseError(new Response("some other 4xx err", { status: 431 }));
```

Comment:
Bad copy/paste? I see five of these tests 👀

Reply:
oh yeah, fixing